src/oxli/kmer_hash.cc

```diff
@@ -146,29 +151,16 @@ std::string _revhash(HashIntoType hash, WordLength k)
 std::string _revcomp(const std::string& kmer)
 {
     std::string out = kmer;
-    size_t ksize = out.size();
-
-    for (size_t i=0; i < ksize; ++i) {
-        char complement;
-
-        switch(kmer[i]) {
-        case 'A':
-            complement = 'T';
-            break;
-        case 'C':
-            complement = 'G';
-            break;
-        case 'G':
-            complement = 'C';
-            break;
-        case 'T':
-            complement = 'A';
-            break;
-        default:
-            complement = kmer[i]; // leave alone.
-            break;
-        }
-        out[ksize - i - 1] = complement;
+
+    auto from = out.begin();
+    auto to = out.end();
+
+    size_t len = to - from;
+    char c;
+    for (to--; from <= to; from++, to--) {
+        c = tbl[(int)*from];
+        *from = tbl[(int)*to];
+        *to = c;
```

Comment:
@luizirber this comes from https://benchmarksgame.alioth.debian.org/u64q/program.php?test=revcomp&lang=gcc&id=2 and speeds things up a lot for me!